scrapy_zyte_api/handler.py

```diff
@@ -86,6 +114,149 @@ def _prepare_api_params(self, request: Request) -> Optional[dict]:
                 f"instead. ({request})."
             )
             raise
+
+        if request.meta.get("zyte_api_automap", self._automap):
```

Comment:
So, the current behavior (writing it down for myself mostly):

Configuration 1: all default options

```py
yield Request(url)  # regular request

yield Request(url, meta={"zyte_api": true})  # Zyte API request with body, headers, method, etc.

# Zyte API request with body, headers, method, etc., but changes the geolocation
yield Request(url, meta={"zyte_api": {"geolocation": "uk"}}) 

# browserHtml requests; some options are not supported (e.g. request method)
yield Request(url, meta={"zyte_api": {"browserHtml": True}}) 

# exact Zyte API request, no magic
yield Request(url, meta={"zyte_api": {"browserHtml": True}, "zyte_api_automap": False}) 

```

Configuration 2: "proxy mode". ZYTE_API_ON_ALL_REQUESTS=True

```py
yield Request(url)  # Zyte API request with body, headers, method, etc.

# Zyte API request with body, headers, method, etc., but with UK geolocation
yield Request(url, meta={"zyte_api": {"geolocation": "uk"}}) 

# browserHtml request; a different mapping of headers (?), less supported Request options
yield Request(url, meta={"zyte_api": {"browserHtml": True}}) 

yield Request(url, meta={"zyte_api": False})  # regular request

# exact Zyte API request, no magic
yield Request(url, meta={"zyte_api": {"browserHtml": True}, "zyte_api_automap": False}) 
```